<?xml version="1.0"?>

<!-- Version 1.1-->
<launch>

  <arg name="Rviz_on" default="true" doc="If true, start Rviz and the joint state gui" />
  <arg name="use_joint_gui" default="true" doc="Should the joint_state_publisher use a GUI for controlling joint states" />

  <arg name="load_default_mia_model" default="false" />

  <arg name="UR_Flange" default = "false" doc="true to visualize the UR flange with the Mia Hand" />

  <!-- Params to set true if you want to rempa the thumb opp joint position  -->
  <arg name="remap_MIA_th_opp" default='false' doc="True to rempa the thumb opp joint position following the indef fle position" />

   <!-- Arguments to upload hardware interfaces of each finger -->
 <arg name="hw_interface_thumb_fle" default="PositionJointInterface"/>
 <arg name="hw_interface_index_fle" default="PositionJointInterface"/>
 <arg name="hw_interface_mrl_fle"   default="PositionJointInterface"/>
 <arg name="robotNamespace"         default="mia_hand"/> <!-- /mia_hand -->



 <group if="$(arg UR_Flange)">
  <!-- Load urd as robot description -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find mia_hand_description)/urdf/mia_hand_onURFlange_launch.urdf.xacro'
  	hw_interface_thumb_fle:=$(arg hw_interface_thumb_fle) hw_interface_index_fle:=$(arg hw_interface_index_fle) hw_interface_mrl_fle:=$(arg hw_interface_mrl_fle)
    robotNamespace:=$(arg robotNamespace)" />
 </group>

  <group unless="$(arg UR_Flange)">
  <!-- Load urd as robot description -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find mia_hand_description)/urdf/mia_hand_launch.urdf.xacro'
    default_mia_model:=$(arg load_default_mia_model)
    hw_interface_thumb_fle:=$(arg hw_interface_thumb_fle)
    hw_interface_index_fle:=$(arg hw_interface_index_fle)
    hw_interface_mrl_fle:=$(arg hw_interface_mrl_fle)
    robotNamespace:=$(arg robotNamespace)" />
 </group>

<!-- Load thumb opp transmission config file-->
 <param name="mia_transmissions_config_" type="string" value="$(find mia_hand_description)/calibration/transmission_config.yaml " />

 <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" ns= "$(arg robotNamespace)">
    <remap from="joint_states" to="miaviz_joint_states" />
  </node>

 <!-- Group of nodes to launch if Rviz_on is true -->
  <group if="$(arg Rviz_on)">

    <!-- Join_state_gui/ publisher -->

    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui"  output="screen" if="$(arg use_joint_gui)">
      <remap from="joint_states" to="$(arg robotNamespace)/joint_states" />
    </node>

     <node name="remap_mia_joint_states" pkg="mia_hand_description" type="remap_mia_joint_states" ns= "$(arg robotNamespace)">
       <param name="remap_MIA_th_opp" type="bool" value="$(arg remap_MIA_th_opp)"/> <!-- private param so as to match to the namespace used-->
      </node>

    <!-- Run rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mia_hand_description)/config/view_mia_hand_config.rviz" required="true">

    </node>

  </group>


</launch>
